### ISSUE TITLE
Restore the site editor navigation panel edit button

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#59372]
### Description

The button to edit navigation links in the Site editor navigation panel has been removed in https://github.com/WordPress/gutenberg/pull/59335

I'm not sure the design feedback and reasoning on https://github.com/WordPress/gutenberg/pull/59335 is ideal.

One of the most confusing and inconsistent things in the Site editor is the mechanism to switch from 'view' mode to 'edit' mode. In a few releases, the affordance of this mechanism has been progressively reduced more and more.
- At some point, there usd to be a blue button with visible text 'Edit'.
- The button was then changed to an icon button with a pencil icon, which is already a cognitive load trigger and an inherent accessibility problem.
- Now hte button in the 'Navigation' section of the panel has been entirely removed.

I'm not sure making the switghing mechanism more obscure and undiscoverable is any progres.

RIght now, the only way to switch to edit mode in this view for keyboard users and screen reader users is to move focus to the iframe preview and press enter. The whole 'clickablity' of the perview iframe is to be hones a big hack and I'd like to see that refactored to a more standard and more semantic UI. I already created a couple issues about it. basically:
- It doesn't work well with screen readers.
- It doesn;t work well with Safari, as the iframe is not exposed in the accessibility tree.
- The focus style on the iframe is little noticeable aend doesn't work at all with Safari.
- Overriding the native role of an iframe with an ARIA `role="button"` is arguably a hack that should be avoided in favor of a standard button.

That said, I kindly disagree with the specific reasoning on https://github.com/WordPress/gutenberg/pull/59335 as well.

I'm not sure the navigation panel should be a place where items can be edited 'in place' in the panel itself in the first place. Additionally, there are other actions that can only be performed in edit mode. For example, renaming a link, adding new links and possibly other actions. 

As such, removing one of the ways to access edit mode seems to me less than ideal.

Also consistency is key. Instead fo making the UI more and more _inconsistent_ I'd like to see an effor to make things more predictable and consistent. In the navigation panel there are other places where an 'edit' icon button is used, for example:
- Templates > A custom template.
- Templates > A default template that has been customized.
- Pages > Any page

I'd rather think that we should _add_ more edit buttons to switch the editor to edit mode rather than removing them

Lastly, I'd like to note I'm not comfortable with the process followed on https://github.com/WordPress/gutenberg/pull/59335. I'm not sure submitting, reviewing, and merging a pull request on a Saturday in a little more than two hours with no accompanying issue to allow for broader discussion (as requested by the contributing guidelines of this project) and not caring abou using any accessibility label is the best way to encourage collaboration and contribution to this project.

### Step-by-step reproduction instructions

- Go to the Site editor > Design > Navigation > Any navigation menu
- In the details view of a menu, observe the edit pencil icon button is no longer there.
- Observe the lack of the edit button is inconsistent with all the other 'details' view like:
  - Templates > A custom template.
  - Templates > A default template that has been customized.
  - Pages > Any page
- Observe the only way to switch the editor to edit mode is now by clicking the preview iframe (or by moving focus to it and pressing Enter).

### Screenshots, screen recording, code snippet

Before https://github.com/WordPress/gutenberg/pull/59335

![before](https://github.com/WordPress/gutenberg/assets/1682452/cb0777b9-1b4d-4287-8343-bea48f4cb925)

After

![after](https://github.com/WordPress/gutenberg/assets/1682452/0ebc0df9-cc78-4160-8747-da19bab2d4ea)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by draganescu** [comment]
Hi @afercia I think the situation now is a bit in-between hence confusing. The end goal as started in #59340 is to remove the edit view for individual navigations. That screen is a stub, it is confusing and there are no plans anywhere to make it better.

That is not to say it doesn't still have a purpose, maybe when Gutenberg natively supports mega menus having a whole canvas for the menu will be something useful.

But today it doesn't offer anything on top of what the navigation block editing offers.

So instead of adding the button back, let's continue on #59340 and just show the home in the frame when we list the available menus?

**Comment by afercia** [comment]
> Hi @afercia I think the situation now is a bit in-between hence confusing. The end goal as started in https://github.com/WordPress/gutenberg/pull/59340 is to remove the edit view for individual navigations. That screen is a stub, it is confusing and there are no plans anywhere to make it better.

I will comment on #59340 on why I think it's not ideal.
